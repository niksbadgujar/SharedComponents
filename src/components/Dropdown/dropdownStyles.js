import styled from "styled-components";
import { Col, Row } from "reactstrap";
import Select, { components } from "react-select";

export const StyledRow = styled(Row)`
  color: black;
`;
export const StyledCol = styled(Col)`
  color: black;
  width: 151px;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
`;

export const StyledHeaderText = styled.text`
  overflow-wrap: break-word !important;
  text-align: left;
  font-size: 16px;
  font-weight: 600;
`;

export const StyledText = styled.text`
  overflow-wrap: break-word !important;
  text-align: left;
  font-size: 14px;
`;

export const StyledLeftTextSection = styled.div`
  width: 20%;
  cursor: pointer;
  margin-left: 1%;
`;

export const StyledAccordionLeftText = styled.div`
  width: 30%;
  margin-left: 1%;
  margin-right: 1%;
`;

export const StyledRightTextSection = styled.div`
  width: fit-content;
  text-align: left;
`;

export const StyledTickIconSection = styled.div`
  width: 5%;
`;

export const StyledImage = styled.img`
  width: 10px;
`;

export const StyledOptionsContainer = styled.div``;

export const StyledArea = styled.div`
  width: 100%;
  position: relative;
`;

export const StyledArrowSection = styled.div`
  width: 10%;
  height: auto;
  position: absolute;
  bottom: 1.2rem;
`;
export const StyledArrowImg = styled.img`
  width: 20px;
`;

export const StyledAccordionRow = styled(Row)`
  margin-top: 1%;
  margin-left: 0%;
  cursor: pointer;
`;

export const StyledSelectContainer = styled.div`
  margin: 8px 0px;
  min-width: 40%;
`;

export const customStyles = {
  control: (providedStyles, state) => ({
    ...providedStyles,
    minHeight: 44,
    minWidth: 223,
    border: "none",
    boxShadow: state.isFocused
      ? "0 0 0 2px black !important"
      : "0 0 0 1px white !important",
    borderRadius: state.isFocused ? "3px" : "0px",
  }),
  valueContainer: (providedStyles) => ({
    ...providedStyles,
    overflow: "none",
  }),
  placeholder: (providedStyles) => ({ ...providedStyles, color: "black" }),
  menuList: (providedStyles) => ({
    ...providedStyles,
    overflowX: "hidden",
    paddingTop: "0px",
    paddingBottom: "0px",
  }),
  menu: (providedStyles) => ({
    ...providedStyles,
    width: "60rem",
    background: "#e6e6e6",
    marginTop: "1px",
    borderTop: "1px solid #b5aca8",
    borderRadius: "0px",
    boxShadow: "none",
    position: "relative",
  }),
  option: (providedStyles, state) => ({
    ...providedStyles,
    // outline: state.isFoused ? "1px soild black !important" : "none",
  }),
  indicatorSeparator: (providedStyles) => ({
    ...providedStyles,
    backgroundColor: "#000000",
    height: "39px",
  }),
};
