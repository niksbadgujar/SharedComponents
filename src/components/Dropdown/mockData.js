export const optionsArray = [
  {
    label: "Hi",
    value: "hi",
    accountNumber: "82622823",
    sortCode: "40-05-20",
    accountType: "Business A/C",
    accountHolderName: "Abc",
    fullName: "abcd, Mr.efg",
    accountHolderType: "ACCOUNT HOLDER",
  },
  {
    label: "Hello",
    value: "hello",
    accountNumber: "5962 9380",
    sortCode: "70-80-60",
    accountType: "Commercial Card",
    accountHolderName: "Luke Co",
    fullName: "Skywalker, Luke/Mr",
    accountHolderType: "CARDHOLDER",
  },
];
