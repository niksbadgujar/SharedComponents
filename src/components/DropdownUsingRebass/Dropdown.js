// @ts-nocheck
import React, { useEffect, useState } from "react";
import styled from "styled-components";
import Select, { components } from "react-select";
import { Box, Flex } from "rebass";
import SingleAccount from "./SingleAccount";
import {
  StyledArrowImg,
  StyledCol, // keep
  StyledImage,
  customStyles,
} from "./dropdownRebassStyles";
import { formatSortCode } from "./Utils/commonMethods";

const CustomOptions = ({ children, ...props }) => {
  const {
    data: {
      sortCode,
      accountHolderName,
      fullName,
      accountNumber,
      accountType,
      accountHolderType,
      balance,
      balanceText,
    } = {},
    isSelected,
    selectProps: {
      value: { accountNumber: defaultAccountNumber = "" } = {},
      accordionClick,
    } = {},
  } = props;

  const [showAccordion, setShowAccordion] = useState(false);

  const StyledOption = styled(components.Option)`
    background: ${isSelected ? "#eceaea !important" : "#eceaea !important"};
    color: #000000 !important;
    border-bottom: 1px solid #cac4c4;
  `;

  const handleRowClick = () => {
    setShowAccordion(!showAccordion);
  };

  const StyledOptionFlex = styled(Flex)`
    padding-bottom: 0.9%;
    border-bottom: ${showAccordion ? "1px solid #cac4c4" : "none"};
    cursor: pointer;
  `;

  const passAccordionData = () => {
    const selectedObj = {
      sortCode,
      accountHolderName,
      fullName,
      accountNumber,
      accountType,
      accountHolderType,
      balance,
      balanceText,
    };
    accordionClick(selectedObj);
  };

  return (
    <>
      <Flex tabIndex={2}>
        <StyledOption {...props}>
          <StyledOptionFlex onClick={handleRowClick}>
            <Box className="col-md-3">
              {formatSortCode(sortCode)} {"  "}
              {accountNumber}
            </Box>

            <Box className="type-holder">
              {accountType} . {accountHolderName}
            </Box>

            {isSelected && accountNumber === defaultAccountNumber && (
              <Box sx={{ width: "5%" }}>
                <StyledImage src="green-tick.png" />
              </Box>
            )}
          </StyledOptionFlex>

          {/* ********************************** Accordion Section ***********************************/}
          {showAccordion && (
            <Flex
              data-testid="accordion-area"
              sx={{ padding: "5px", position: "relative" }}
            >
              <Box
                sx={{ width: "10%", position: "absolute", bottom: "1.2rem" }}
              >
                <StyledArrowImg src="down-arrow.png" width="25px" />
              </Box>

              <Flex
                className="col-md-7"
                onClick={passAccordionData}
                sx={{ justifyContent: "space-evenly", cursor: "pointer" }}
              >
                <Box>
                  {formatSortCode(sortCode)}
                  {"  "}
                  {accountNumber}
                </Box>

                <Box>
                  {accountHolderType} . {fullName}
                </Box>
              </Flex>
            </Flex>
          )}
        </StyledOption>
      </Flex>
    </>
  );
};

const formatOptionLabel = (props) => {
  const { accountType, accountHolderName, accountNumber, sortCode } = props;
  return (
    <>
      <SingleAccount
        accountType={accountType}
        accountHolderName={accountHolderName}
        accountNumber={accountNumber}
        sortCode={sortCode}
        isSingleAccount={false}
      />
    </>
  );
};

const Dropdown = (props) => {
  const { options = [], defaultAccount = {} } = props;
  const [selectedAccount, setSelectedAccount] = useState(defaultAccount);

  useEffect(() => {
    const { selectedAccount } = props;
    setSelectedAccount(defaultAccount);
    selectedAccount(defaultAccount);
  }, [defaultAccount]);

  // const handleSelect = (selectedAccount) => {
  //   const {
  //     accountType,
  //     accountHolderName,
  //     accountNumber,
  //     sortCode,
  //   } = selectedAccount;
  //   const selectedAccountObj = {
  //     accountType,
  //     accountHolderName,
  //     accountNumber,
  //     sortCode,
  //   };
  //   setSelectedAccount(selectedAccountObj);
  // };

  const {
    sortCode,
    accountHolderName,
    accountNumber,
    accountType,
  } = selectedAccount;

  const handleAccordionClick = (selectedObj) => {
    const { selectedAccount } = props;
    const {
      accountType,
      accountHolderName,
      accountNumber,
      sortCode,
      balance,
      balanceText,
    } = selectedObj;
    const selectedAccountObj = {
      accountType,
      accountHolderName,
      accountNumber,
      sortCode,
      balance,
      balanceText,
    };
    setSelectedAccount(selectedAccountObj);
    selectedAccount(selectedAccountObj);
  };

  return (
    <>
      <Select
        id="select"
        className="select"
        options={options}
        components={{ Option: CustomOptions }}
        formatOptionLabel={formatOptionLabel}
        styles={customStyles}
        defaultValue={{
          sortCode,
          accountNumber,
          accountHolderName,
          accountType,
        }}
        // onChange={(selectedAccount) => handleSelect(selectedAccount)}
        value={selectedAccount}
        closeMenuOnSelect={false}
        isSearchable={false}
        accordionClick={handleAccordionClick}
        // menuIsOpen
        tabIndex={"0"}
        tabSelectsValue
      />
    </>
  );
};

export default Dropdown;
