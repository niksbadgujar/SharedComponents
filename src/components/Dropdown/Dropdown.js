// @ts-nocheck
import React, { useEffect, useRef, useState } from "react";
import styled from "styled-components";
import Select, { components } from "react-select";
import { Col, Row } from "reactstrap";
import { optionsArray } from "./mockData";
import {
  StyledAccordionLeftText,
  StyledOptionsContainer,
  StyledAccordionRow,
  StyledArea,
  StyledArrowImg,
  StyledArrowSection,
  StyledCol,
  StyledHeaderText,
  StyledImage,
  StyledLeftTextSection,
  StyledRightTextSection,
  StyledRow,
  StyledSelectContainer,
  StyledText,
  StyledTickIconSection,
  customStyles,
} from "./dropdownStyles";

const CustomOptions = React.memo(({ children, ...props }) => {
  const {
    data: {
      sortCode,
      accountHolderName,
      fullName,
      accountNumber,
      accountType,
      accountHolderType,
    } = {},
    isSelected,
    selectProps: {
      value: { accountNumber: defaultAccountNumber = "" } = {},
      accordionClick,
    } = {},
  } = props;

  console.log("props - ", props);
  const [showAccordion, setShowAccordion] = useState(false);

  const StyledOption = styled(components.Option)`
    background: ${isSelected ? "#eceaea !important" : "#eceaea !important"};
    color: #000000 !important;
    border-bottom: 1px solid #cac4c4;
  `;

  const handleRowClick = () => {
    setShowAccordion(!showAccordion);
  };

  const StyledOptionRow = styled(Row)`
    padding-bottom: 0.9%;
    border-bottom: ${showAccordion ? "1px solid #cac4c4" : "none"};
  `;

  const passAccordionData = () => {
    const selectedObj = {
      sortCode,
      accountHolderName,
      fullName,
      accountNumber,
      accountType,
      accountHolderType,
    };
    accordionClick(selectedObj);
  };

  return (
    <>
      <StyledOptionsContainer>
        <StyledOption {...props}>
          <StyledOptionRow>
            <StyledLeftTextSection onClick={handleRowClick}>
              {sortCode} {accountNumber}
            </StyledLeftTextSection>

            <StyledRightTextSection>
              {accountType} . {accountHolderName}
            </StyledRightTextSection>

            {isSelected && accountNumber === defaultAccountNumber && (
              <StyledTickIconSection>
                <StyledImage src="green-tick.png" />
              </StyledTickIconSection>
            )}
          </StyledOptionRow>

          {/* ********************************** Accordion Section ***********************************/}

          {showAccordion && (
            <StyledArea data-testid="accordion-area">
              <StyledArrowSection>
                <StyledArrowImg src="down-arrow.png" width="25px" />
              </StyledArrowSection>

              <StyledAccordionRow onClick={passAccordionData}>
                <StyledAccordionLeftText>
                  {sortCode} {accountNumber}
                </StyledAccordionLeftText>

                <StyledRightTextSection>
                  {accountHolderType} . {fullName}
                </StyledRightTextSection>
              </StyledAccordionRow>
            </StyledArea>
          )}
        </StyledOption>
      </StyledOptionsContainer>
    </>
  );
});

const formatOptionLabel = (props) => {
  const { ...rest } = props.innerProps;
  const newProps = Object.assign(props, { innerProps: rest });
  const {
    value,
    accountType,
    accountHolderName,
    accountNumber,
    sortCode,
  } = newProps;

  if (value === "") {
    return (
      <StyledRow>
        <StyledText>Please Select</StyledText>
      </StyledRow>
    );
  } else {
    return (
      <StyledRow>
        <StyledCol>
          <StyledRow>
            <StyledCol>
              <StyledHeaderText>
                {sortCode} {accountNumber}
              </StyledHeaderText>
            </StyledCol>
          </StyledRow>

          <StyledRow>
            <StyledCol>
              <StyledText>
                {accountType} . {accountHolderName}
              </StyledText>
            </StyledCol>
          </StyledRow>
        </StyledCol>
      </StyledRow>
    );
  }
};

const Dropdown = React.memo((props) => {
  const StyledSelect = styled(Select)``;

  const [options, setOptions] = useState([]);
  const [selectedAccount, setSelectedAccount] = useState({});

  useEffect(() => {
    getData();
  }, []);

  const getData = () => {
    const accounts = optionsArray;
    const modifiedAccountsArray = accounts.map((obj) => ({
      sortCode: obj.sortCode,
      accountHolderName: obj.accountHolderName,
      accountNumber: obj.accountNumber,
      accountType: obj.accountType,
      fullName: obj.fullName,
      accountHolderType: obj.accountHolderType,
    }));
    setSelectedAccount(modifiedAccountsArray[0]);
    setOptions(modifiedAccountsArray);
  };

  const handleSelect = (selectedAccount) => {
    const {
      accountType,
      accountHolderName,
      accountNumber,
      sortCode,
    } = selectedAccount;
    const selectedAccountObj = {
      accountType,
      accountHolderName,
      accountNumber,
      sortCode,
    };
    // setSelectedAccount(selectedAccountObj);
  };

  const defaultValue = options[0] || {};
  const {
    sortCode,
    accountHolderName,
    accountNumber,
    accountType,
  } = defaultValue;

  const handleAccordionClick = (selectedObj) => {
    const {
      accountType,
      accountHolderName,
      accountNumber,
      sortCode,
    } = selectedObj;
    const selectedAccountObj = {
      accountType,
      accountHolderName,
      accountNumber,
      sortCode,
    };
    setSelectedAccount(selectedAccountObj);
  };

  return (
    <>
      <Row className="mt-5 ml-5">
        <Col md={2}>
          <StyledSelectContainer>
            <StyledSelect
              options={options}
              components={{ Option: CustomOptions }}
              formatOptionLabel={formatOptionLabel}
              styles={customStyles}
              defaultValue={{
                sortCode,
                accountNumber,
                accountHolderName,
                accountType,
              }}
              // onChange={(selectedAccount) => handleSelect(selectedAccount)}
              value={selectedAccount}
              closeMenuOnSelect={false}
              isSearchable={false}
              accordionClick={handleAccordionClick}
              // menuIsOpen
              // tabSelectsValue={false}
            />
          </StyledSelectContainer>
        </Col>
      </Row>
      <Row>
        <Col md={2}>Hello</Col>
      </Row>
    </>
  );
});

export default Dropdown;
