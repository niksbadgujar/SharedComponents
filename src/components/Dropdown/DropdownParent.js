import Dropdown from "./Dropdown";

const DropdownParent = (props) => {
  return (
    <>
      <Dropdown />
    </>
  );
};

export default DropdownParent;
