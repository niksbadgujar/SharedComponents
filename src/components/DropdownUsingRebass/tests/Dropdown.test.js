import { render, cleanup } from "@testing-library/react";
import Dropdown from "../Dropdown";

describe("Test suite for Dropdown component", () => {
  const options = [
    {
      accountType: "Business A/C",
      accountHolderName: "Abc",
      accountNumber: "82622823",
      sortCode: "400520",
      balance: "-1898.55",
      balanceText: "GBL",
    },
    {
      accountType: "Business A/C",
      accountHolderName: "Abc",
      accountNumber: "82622897",
      sortCode: "800523",
      balance: "188.55",
      balanceText: "GBL",
    },
  ];
  const { container, getByTestId, getByRole } = render(
    <Dropdown
      options={options}
      defaultAccount={options[0]}
      isSingleAccount={false}
      accountType="Business A/C"
      accountHolderName="ABC"
      accountNumber="82622823"
      sortCode="400520"
      selectedAccount={jest.fn()}
    />
  );

  test("Snapshot test to check component renders properly ", () => {
    expect(container).toMatchSnapshot();
  });

  test("Select Dropdown", () => {
    console.log("", container);
  });
  // test("Test", () => {
  //   console.log("child - ", container.getElementsByClassName("select"));
  // });

  // test("Test to check component renders", () => {
  //   const { container, getByTestId, getByRole } = render(
  //     <Dropdown
  //       options={options}
  //       defaultAccount={options[0]}
  //       isSingleAccount={false}
  //       accountType="Business A/C"
  //       accountHolderName="ABC"
  //       accountNumber="82622823"
  //       sortCode="400520"
  //       selectedAccount={jest.fn()}
  //     />
  //   );
  //   expect(container.querySelector("#sort-code").textContent).toBe(
  //     "40-05-20  82622823"
  //   );
  // });
});
