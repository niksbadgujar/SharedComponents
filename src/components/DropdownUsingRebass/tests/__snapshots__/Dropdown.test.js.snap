// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for Dropdown component Snapshot test to check component renders properly  1`] = `
<div>
  <div
    class="select css-2b097c-container"
    id="select"
  >
    <div
      class=" css-2d0fx8-control"
    >
      <div
        class=" css-1p7dxgv-ValueContainer"
      >
        <div
          class=" css-1uccc91-singleValue"
        >
          <div
            class="box-class css-1py8xe8"
          >
            <div
              class="css-4cffwv"
            >
              <div
                class="css-vurnku"
              >
                <span
                  class="sc-bdfBwQ gIeWGu"
                  id="header-text"
                >
                  40-05-20
                    
                  82622823
                </span>
              </div>
            </div>
            <div
              class="css-4cffwv"
            >
              <div
                class="css-vurnku"
              >
                <span
                  class="sc-gsTCUz VbRzm"
                >
                  Business A/C
                   . 
                  Abc
                </span>
              </div>
            </div>
          </div>
        </div>
        <input
          aria-autocomplete="list"
          class="css-62g3xt-dummyInput"
          id="react-select-2-input"
          readonly=""
          tabindex="0"
          value=""
        />
      </div>
      <div
        class=" css-1hb7zxy-IndicatorsContainer"
      >
        <span
          class=" css-1lh389c-indicatorSeparator"
        />
        <div
          aria-hidden="true"
          class=" css-tlfecz-indicatorContainer"
        >
          <svg
            aria-hidden="true"
            class="css-6q0nyr-Svg"
            focusable="false"
            height="20"
            viewBox="0 0 20 20"
            width="20"
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
